spec/reporter_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe 'Wprof Generate Profiling' do
+  require 'spec_helper'
+  context 'When I try to gen a Report' do
+    context 'with Standard rec_type' do
+      before do
+        allow(GenProf).to receive(:deploy_reporter)
+      end
+      # params_exp = %i[transaction_id total_time start_dt end_dt code controller url db_runtime]
+      let(:data) do
+        new_gen_prof = GenProf.new(Event.new, :standard)
+        new_gen_prof.instance_variable_get(:@params)
+      end
+      let(:rec_type) { :standard }
+      let(:spect_param) do
+        %i[transaction_id total_time start_dt end_dt code controller url db_runtime]
+      end
+
+      it 'And reporter Type wasnt defined' do
+        allow(WProf::Config).to receive_messages(get_value: '')
+        record = WprofReporter.new.perform(data, rec_type)
+        expect(record.keys).to eq(spect_param)
+      end
+
+      it 'And reporter Type is LOGGER' do
+        allow(WProf::Config).to receive_messages(get_value: 'LOGGER')
```

Comment:
This leads me to think that LOGGER/DATABASE and such should be constants in some class in the gem. Is that possible?

Reply:
They aren't constants, they are string options writing for the User. Could be a group of values for an unique constant, but these validation not exist yet.